### PR TITLE
statusCode check added

### PR DESCRIPTION
**Pull request by 1lastBr3ath**
Please check and verify it on your end.

### CODE REVIEW

File: DetectDynamicJS.py [hunk 1]
```diff
@@ -70,6 +70,14 @@ def doPassiveScan(self, baseRequestResponse):
         self._requestResponse = baseRequestResponse
         self._helpers = self._callbacks.getHelpers()
         
+        response = self._requestResponse.getResponse()
+        responseInfo = self._helpers.analyzeResponse(response)
+        
+        # Scan only if the statusCode is 200
+        statusCode = responseInfo.getStatusCode()
+        if statusCode != 200:
+          return
```
Comment (luh2) [inline DetectDynamicJS.py]: should be using indentation level of 4 characters

File: DetectDynamicJS.py [hunk 2]
```diff
@@ -154,15 +159,20 @@ def isScript(self, requestResponse):
             if len(contentTypeL) == 1:
                 contentType = contentTypeL[0].lower()
             statusCode = responseInfo.getStatusCode()
-            if (any(content in contentType for content in possibleContentTypes) or "script" in inferredMimeType or "script" in mimeType) and (int(statusCode) < 300 or int(statusCode) > 399):
-                return True
```
Comment (luh2) [inline DetectDynamicJS.py]: Indentation level of 4 is needed.

File: DetectDynamicJS.py [hunk 3]
```diff
@@ -154,15 +159,20 @@ def isScript(self, requestResponse):
...
+              return True
         return False
         
 
     def compareResponses(self, newResponse, oldResponse):
         """Compare two responses in respect to their body contents"""
         nResponse = newResponse.getResponse()
         nResponseInfo = self._helpers.analyzeResponse(nResponse)
+        
+        # Check if the statusCode is still 200
```
Comment (luh2) [inline DetectDynamicJS.py]: For reducing false positives, this is not necessary. If you get a 200 on the first request and a 302 or something afterwards you get a login oracle. So it actually creates false negatives.

File: DetectDynamicJS.py [hunk 4]
```diff
@@ -77,7 +77,14 @@ def doPassiveScan(self, baseRequestResponse):
...
         if statusCode != 200:
           return
-          
+
+        # Check for authorization
+        reqHeaders = self._helpers.analyzeRequest(self._requestResponse).getHeaders()
+        hfields = [h.split(':')[0] for h in reqHeaders]
+        ifields = ['cookie','authorization']
+        if not any(h for h in ifields if h not in str(hfields).lower()):
+          return
```
Comment (luh2) [inline DetectDynamicJS.py]: I like this check. But please use `return None` in this function. It is a style question. Also indentation again.

File: DetectDynamicJS.py [hunk 5]
```diff
@@ -77,6 +77,11 @@ def doPassiveScan(self, baseRequestResponse):
         statusCode = responseInfo.getStatusCode()
         if statusCode != 200:
           return
+        
+        # Check if the script is world readable
```
Comment (luh2) [inline DetectDynamicJS.py]: Not sure why you added this check. In which case would that change something? Scripts can always be included x-domain.

File: DetectDynamicJS.py (hunk 1 above)
Comment (luh2) [inline DetectDynamicJS.py]: Oh, and also please use `return None` inside this function for style and consistency reasons.

File: DetectDynamicJS.py (hunk 5 above)
Comment (1lastBr3ath) [inline DetectDynamicJS.py]: As I said previously, browsers cannot read x-domain resources or scripts (in most cases) if they respond with status code other than 200. Please try it yourself :)

Comment (luh2) [inline DetectDynamicJS.py]: I am. Just give me a few weeks. I'm not full-time working on this, will have to find some free time first.

Comment (1lastBr3ath) [inline DetectDynamicJS.py]: Same here :)
Found them while doing a pentest, in real environment. Jus wanted to update things on your end.

File: DetectDynamicJS.py [hunk 6]
```diff
@@ -154,15 +172,16 @@ def isScript(self, requestResponse):
             if len(contentTypeL) == 1:
                 contentType = contentTypeL[0].lower()
             statusCode = responseInfo.getStatusCode()
-            if (any(content in contentType for content in possibleContentTypes) or "script" in inferredMimeType or "script" in mimeType) and (int(statusCode) < 300 or int(statusCode) > 399):
-                return True
+            if (any(content in contentType for content in possibleContentTypes) or "script" in inferredMimeType or "script" in mimeType):
+              return True
```
Comment (luh2) [inline DetectDynamicJS.py]: As soon as the right indendation is used, i'm happy to merge this. Nice additions!

File: DetectDynamicJS.py [hunk 7]
```diff
@@ -70,6 +70,26 @@ def doPassiveScan(self, baseRequestResponse):
...
+        response = self._requestResponse.getResponse()
+        responseInfo = self._helpers.analyzeResponse(response)
+        
+        # Scan only if the statusCode is 200
+        statusCode = responseInfo.getStatusCode()
+        if statusCode != 200:
+            return None
+        
+        # Check if the script is world readable
+        resHeaders = self._helpers.analyzeResponse(response).getHeaders()
```
Comment (luh2) [inline DetectDynamicJS.py]: This block I don't want for now. Might add it later.

Comment (1lastBr3ath) [inline DetectDynamicJS.py]: I don't understand why you want it removed. You previously agreed to check statusCode, and said it's nice. Anyway, I'll remove it and commit as requested.

Comment (luh2) [inline DetectDynamicJS.py]: It was for acces-control-allow-origin header, not for the status code. You are right, the status code I do want. Sorry, I thought it was obvious, I'll try to be more exact on what I mean next time.

**Pull request merged**